snapcraft/yaml.py

```diff
@@ -247,19 +247,23 @@ def build_env(self, root):
         env.append('PERL5LIB={0}/usr/share/perl5/'.format(root))
         return env
 
-    def build_env_for_part(self, part):
-        # Grab build env of all part's dependencies
+    def build_env_for_part(self, part, dep=False):
```

Comment:
I think we're over using the term 'dep' in this function. Perhaps rename this to 'first' or 'root' or 'recurse', I was confused at first why we were checking dependencies if dep=False.